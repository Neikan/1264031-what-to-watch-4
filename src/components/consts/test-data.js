export const Promo = {
  TITLE: `The Grand Budapest Hotel poster`,
  GENRE: `Drama`,
  DATE: 2014
};

export const MOVIES = [
  {
    id: `mock-1`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald`
  }, {
    id: `mock-2`,
    title: `Bohemian Rhapsody`,
    poster: `bohemian-rhapsody`
  }, {
    id: `mock-3`,
    title: `Macbeth`,
    poster: `macbeth`
  }
];

