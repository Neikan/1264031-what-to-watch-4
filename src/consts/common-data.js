export const ALL_GENRES = `All genres`;


export const AuthStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};


export const BASE_URL = `https://4.react.pages.academy`;


export const COUNT_GENRES = 10;


export const CountMovies = {
  ALL: 20,
  START: 8,
  LIKED_BY_GENRE: 4
};


export const DECIMAL_BASE = 10;


export const FormatRule = {
  REVIEW_DATE: `MMMM DD, YYYY`,
  REVIEW_DATE_HIDE: `YYYY-MM-DD`
};


export const HIDDEN_CLASS = `visually-hidden`;


export const LogoPosition = {
  HEADER: `header`,
  FOOTER: `footer`
};


export const MovieTabList = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};


export const PAD_STRING_ZERO = `0`;


export const Page = {
  MAIN: `/`,
  MOVIE: `films`,
  LOGIN: `login`,
  PLAYER: `player`,
  ADD_REVIEW: `review`
};


export const Rating = {
  DEFAULT: 3,
  WEIGHT: 2
};


export const RATING_STARS = [1, 2, 3, 4, 5];


export const ReviewMessage = {
  DEFAULT: ``,
  LENGTH_MIN: 50,
  LENGTH_MAX: 400
};


export const ReviewMessageStatus = {
  DEFAULT: ``,
  SUCCESS: `Ваш комментарий добавлен!`,
  ERROR_SENDING: `Извините, при добавлении Вашего комментария возникла ошибка. Попробуйте позднее.`,
  MESSAGE_LENGTH: `Текущая длина сообщения:`
};


export const REVIEW_PLACEHOLDER = `Введите сообщение не менее ${ReviewMessage.LENGTH_MIN} и не более ${ReviewMessage.LENGTH_MAX} символов`;


export const RequestStatusCode = {
  OK: 200,
  UNAUTHORIZED: 401
};


export const TextualRating = {
  BAD: {
    level: `Bad`,
    from: 0,
  },
  NORMAL: {
    level: `Normal`,
    from: 3
  },
  GOOD: {
    level: `Good`,
    from: 5
  },
  VERY_GOOD: {
    level: `Very good`,
    from: 8
  },
  AWESOME: {
    level: `Awesome`,
    from: 10
  }
};


export const Time = {
  MINUTES: 60,
  SECONDS: 60
};


export const Timeout = {
  WAIT: 2000,
  REQUEST: 5000
};


export const Url = {
  WTW: `wtw`,
  FILMS: `films`,
  COMMENTS: `comments`,
  PROMO: `promo`,
  LOGIN: `login`
};
