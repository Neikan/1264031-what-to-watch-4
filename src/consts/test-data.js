const MOVIE_POSTER = `the-grand-budapest-hotel-poster`;

export const LogoPosition = {
  HEADER: `header`,
  FOOTER: `footer`
};


export const MOVIES = [
  {
    id: `mock-1`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: MOVIE_POSTER,
    cover: `fantastic-beasts-the-crimes-of-grindelwald`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/f/fd/%22Night_After_Night%22_Trailer_1932.webm/%22Night_After_Night%22_Trailer_1932.webm.480p.vp9.webm`,
    rating: {
      score: 4.6,
      level: `Normal`,
      countVotes: 240,
    },
    director: `Крис МакКей`,
    actors: `Крис О’Доннелл, Джек Николсон`,
    duration: 120,
    year: 2015,
    genre: `Documentary`,
    description: `Восемь лет назад Бэтмен растворился в ночи, превратившись из героя в беглеца.`,
    reviews: []
  }, {
    id: `mock-2`,
    title: `Bohemian Rhapsody`,
    poster: MOVIE_POSTER,
    cover: `bohemian-rhapsody`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/c/cd/12_Angry_Men_%281957%29_-_Trailer.webm/12_Angry_Men_%281957%29_-_Trailer.webm.480p.vp9.webm`,
    rating: {
      score: 9.1,
      level: `Very good`,
      countVotes: 10,
    },
    director: `Крис МакКей`,
    actors: `Арнольд Шварценеггер, Кристиан Бэйл`,
    duration: 320,
    year: 1999,
    genre: `Crime`,
    description: `Бэтмен поднимает ставки в войне с криминалом. С помощью лейтенанта Джима Гордона и прокурора Харви Дента он намерен очистить улицы от преступности, отравляющей город. Сотрудничество оказывается эффективным, но скоро они обнаружат себя посреди хаоса, развязанного восходящим криминальным гением, известным испуганным горожанам под именем Джокер.`,
    reviews: [{
      id: `reviews-9`,
      text: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
      author: `Lickona Matthew`,
      date: `December 2, 2016`,
      rating: 1.2
    }]
  }, {
    id: `mock-3`,
    title: `Macbeth`,
    poster: MOVIE_POSTER,
    cover: `macbeth`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/c/ca/2001_A_Space_Odyssey_%281968%29_-_Trailer.webm/2001_A_Space_Odyssey_%281968%29_-_Trailer.webm.360p.vp9.webm`,
    rating: {
      score: 6.0,
      level: `Good`,
      countVotes: 1240,
    },
    director: `Джоэл Шумахер`,
    actors: ` Ким Бейсингер, Том Харди, Кэнта Миякэ, Марион Котийяр`,
    duration: 20,
    year: 2020,
    genre: `Horror`,
    description: `Приняв на себя вину за смерть прокурора Харви Дента, он пожертвовал всем. Вместе с комиссаром Гордоном они решили, что так будет лучше для всех.`,
    reviews: [{
      id: `reviews-1`,
      text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
      author: `Kate Muir`,
      date: `December 24, 2016`,
      rating: 8.9
    }, {
      id: `reviews-2`,
      text: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
      author: `Bill Goodykoontz`,
      date: `November 18, 2015`,
      rating: 5.0
    }, {
      id: `reviews-3`,
      text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
      author: `Amanda Greever`,
      date: `November 18, 2015`,
      rating: 3.3
    }, {
      id: `reviews-4`,
      text: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
      author: `Matthew Lickona`,
      date: `December 20, 2016`,
      rating: 7.2
    }, {
      id: `reviews-5`,
      text: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
      author: `Paula Fleri-Soler`,
      date: `December 20, 2016`,
      rating: 10
    }]
  }
];


export const GENRES = [
  {
    id: `g1`,
    title: `All genres`,
    isActive: true
  }, {
    id: `g2`,
    title: `Comedies`,
    isActive: false
  }, {
    id: `g3`,
    title: `Crime`,
    isActive: false
  }
];


export const MarkupElement = {
  PROMO_BTN_PLAY: `btn--play`,
  PROMO_BTN_ADD_TO_LIST: `btn--list`,
  MOVIE_CARD: `small-movie-card`,
  MOVIE_CARD_TITLE: `small-movie-card__title`,
  MOVIE_CARD_COVER: `small-movie-card__image`,
  GENRE_LINK: `catalog__genres-link`,
  MOVIE_TAB: `movie-nav__item`,
  BTN_SHOW_MORE: `catalog__button`
};
