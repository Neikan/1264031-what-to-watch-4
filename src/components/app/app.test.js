// Импорт библиотек
import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

// Импорт компонентов
import App from "./app.jsx";

// Импорт типов, констант, утилит
import {Page, CountMovies, MOVIES, GENRES, UserDatumState, UserDatumStateNoAuth} from "./../../consts/test-data";
import {getLikedMoviesByGenre, getMoviesByGenre} from "../../utils/common.js";
import NameSpace from "../../store/name-space.js";


const mockStore = configureStore([]);


describe(`Test App component`, () => {
  test(`App component is created and rendered correctly when page is Main`, () => {
    const store = mockStore({
      [NameSpace.DATUM]: {
        page: Page.MAIN,

        movies: MOVIES,
        promoMovie: MOVIES[0],
        selectedMovie: MOVIES[0],
        likedMovies: getLikedMoviesByGenre(MOVIES, MOVIES[0].genre, MOVIES[0].id),
        countShowedMovies: CountMovies.START,
        isPlayingMovie: false,

        genres: GENRES,
        selectedGenre: GENRES[0].title
      },
      [NameSpace.DATUM_USER]: UserDatumState
    });


    const tree = renderer.create(
        <Provider store={store}>
          <App />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });


  test(`App component is created and rendered correctly when page is Movie`, () => {
    const store = mockStore({
      [NameSpace.DATUM]: {
        page: Page.MOVIE,

        movies: MOVIES,
        promoMovie: MOVIES[0],
        selectedMovie: MOVIES[1],
        likedMovies: getLikedMoviesByGenre(MOVIES, MOVIES[1].genre, MOVIES[1].id),
        countShowedMovies: CountMovies.START,
        isPlayingMovie: false,

        genres: GENRES,
        selectedGenre: GENRES[0].title
      },
      [NameSpace.DATUM_USER]: UserDatumState
    });


    const tree = renderer.create(
        <Provider store={store}>
          <App />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });


  test(`App component is created and rendered correctly when select Genre`, () => {
    const store = mockStore({
      [NameSpace.DATUM]: {
        page: Page.MAIN,

        movies: MOVIES,
        promoMovie: MOVIES[0],
        selectedMovie: MOVIES[0],
        likedMovies: getMoviesByGenre(MOVIES, GENRES[1]),
        countShowedMovies: CountMovies.START,
        isPlayingMovie: false,

        genres: GENRES,
        selectedGenre: GENRES[1].title
      },
      [NameSpace.DATUM_USER]: UserDatumState
    });


    const tree = renderer.create(
        <Provider store={store}>
          <App />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });


  test(`App component is created and rendered correctly when promoMovie === null`, () => {
    const store = mockStore({
      [NameSpace.DATUM]: {
        page: Page.MAIN,

        movies: MOVIES,
        promoMovie: null,
        selectedMovie: MOVIES[0],
        likedMovies: getMoviesByGenre(MOVIES, GENRES[1]),
        countShowedMovies: CountMovies.START,
        isPlayingMovie: false,

        genres: GENRES,
        selectedGenre: GENRES[1].title
      },
      [NameSpace.DATUM_USER]: UserDatumState
    });


    const tree = renderer.create(
        <Provider store={store}>
          <App />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });


  test(`App component is created and rendered correctly when authStatus === NO_AUTH`, () => {
    const store = mockStore({
      [NameSpace.DATUM]: {
        page: Page.MAIN,

        movies: MOVIES,
        promoMovie: MOVIES[0],
        selectedMovie: MOVIES[0],
        likedMovies: getMoviesByGenre(MOVIES, GENRES[1]),
        countShowedMovies: CountMovies.START,
        isPlayingMovie: false,

        genres: GENRES,
        selectedGenre: GENRES[1].title
      },
      [NameSpace.DATUM_USER]: UserDatumStateNoAuth
    });


    const tree = renderer.create(
        <Provider store={store}>
          <App />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
