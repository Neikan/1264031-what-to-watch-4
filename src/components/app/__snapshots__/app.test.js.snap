// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component App component is created and rendered correctly when authStatus === NO_AUTH 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom3.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
          onClick={[Function]}
        >
          <img
            alt="Fantastic Beasts: The Crimes of Grindelwald"
            height="327"
            src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom1.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
            onClick={[Function]}
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Documentary
            </span>
            <span
              className="movie-card__year"
            >
              2015
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="/"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="/"
            onClick={[Function]}
          >
            Comedies
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="/"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
      </ul>
      <div>
        <div
          className="catalog__movies-list"
        />
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`Test App component App component is created and rendered correctly when page is Main 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom3.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="Batman"
            height="63"
            src="https://4.react.pages.academy/wtw/static/avatar/5.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
          onClick={[Function]}
        >
          <img
            alt="Fantastic Beasts: The Crimes of Grindelwald"
            height="327"
            src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom1.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
            onClick={[Function]}
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Documentary
            </span>
            <span
              className="movie-card__year"
            >
              2015
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="/"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="/"
            onClick={[Function]}
          >
            Comedies
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="/"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
      </ul>
      <div>
        <div
          className="catalog__movies-list"
        >
          <article
            className="small-movie-card catalog__movies-card"
            id="mock-1"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="175"
                poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom2.jpg"
                type="video/webm"
                width="280"
              >
                Sorry, your browser doesn\`t support embedded videos
              </video>
            </div>
            <h3
              className="small-movie-card__title"
              onClick={[Function]}
            >
              <a
                className="small-movie-card__link"
                href="/films/mock-1"
                onClick={[Function]}
              >
                Fantastic Beasts: The Crimes of Grindelwald
              </a>
            </h3>
          </article>
          <article
            className="small-movie-card catalog__movies-card"
            id="mock-2"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="175"
                poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york2.jpg"
                type="video/webm"
                width="280"
              >
                Sorry, your browser doesn\`t support embedded videos
              </video>
            </div>
            <h3
              className="small-movie-card__title"
              onClick={[Function]}
            >
              <a
                className="small-movie-card__link"
                href="/films/mock-2"
                onClick={[Function]}
              >
                Bohemian Rhapsody
              </a>
            </h3>
          </article>
          <article
            className="small-movie-card catalog__movies-card"
            id="mock-3"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="175"
                poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/pulp-fiction2.jpg"
                type="video/webm"
                width="280"
              >
                Sorry, your browser doesn\`t support embedded videos
              </video>
            </div>
            <h3
              className="small-movie-card__title"
              onClick={[Function]}
            >
              <a
                className="small-movie-card__link"
                href="/films/mock-3"
                onClick={[Function]}
              >
                Macbeth
              </a>
            </h3>
          </article>
        </div>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`Test App component App component is created and rendered correctly when page is Movie 1`] = `
<section
  className="movie-card movie-card--full"
  style={
    Object {
      "backgroundColor": "#333",
    }
  }
>
  <div
    className="movie-card__hero"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Bohemian Rhapsody"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york3.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="Batman"
            height="63"
            src="https://4.react.pages.academy/wtw/static/avatar/5.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          Bohemian Rhapsody
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Crime
          </span>
          <span
            className="movie-card__year"
          >
            1999
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <button
            className="btn btn--play movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </button>
          <button
            className="btn btn--list movie-card__button"
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
          <a
            className="btn movie-card__button"
            href="/review"
            onClick={[Function]}
          >
            Add review
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="movie-card__wrap movie-card__translate-top"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster movie-card__poster--big"
      >
        <img
          alt="Bohemian Rhapsody"
          height="327"
          src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york1.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <nav
          className="movie-nav movie-card__nav"
        >
          <ul
            className="movie-nav__list"
          >
            <li
              className="movie-nav__item movie-nav__item--active"
              onClick={[Function]}
            >
              <a
                className="movie-nav__link"
                href="/"
                onClick={[Function]}
              >
                Overview
              </a>
            </li>
            <li
              className="movie-nav__item"
              onClick={[Function]}
            >
              <a
                className="movie-nav__link"
                href="/"
                onClick={[Function]}
              >
                Details
              </a>
            </li>
            <li
              className="movie-nav__item"
              onClick={[Function]}
            >
              <a
                className="movie-nav__link"
                href="/"
                onClick={[Function]}
              >
                Reviews
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="movie-card__text movie-card__row"
        >
          <div
            className="movie-card__text-col"
          >
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Director
              </strong>
              <span
                className="movie-card__details-value"
              >
                ÐÑÐ¸Ñ ÐÐ°ÐºÐÐµÐ¹
              </span>
            </p>
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Starring
              </strong>
              <span
                className="movie-card__details-value"
              >
                ÐÑÐ½Ð¾Ð»ÑÐ´ Ð¨Ð²Ð°ÑÑÐµÐ½ÐµÐ³Ð³ÐµÑ, ÐÑÐ¸ÑÑÐ¸Ð°Ð½ ÐÑÐ¹Ð»
              </span>
            </p>
          </div>
          <div
            className="movie-card__text-col"
          >
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Run Time
              </strong>
              <span
                className="movie-card__details-value"
              >
                320
              </span>
            </p>
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Genre
              </strong>
              <span
                className="movie-card__details-value"
              >
                Crime
              </span>
            </p>
            <p
              className="movie-card__details-item"
            >
              <strong
                className="movie-card__details-name"
              >
                Released
              </strong>
              <span
                className="movie-card__details-value"
              >
                1999
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Test App component App component is created and rendered correctly when promoMovie === null 1`] = `
<section>
  <h2
    className="loader"
  >
    Loading... Please wait
  </h2>
</section>
`;

exports[`Test App component App component is created and rendered correctly when select Genre 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom3.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="Batman"
            height="63"
            src="https://4.react.pages.academy/wtw/static/avatar/5.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
          onClick={[Function]}
        >
          <img
            alt="Fantastic Beasts: The Crimes of Grindelwald"
            height="327"
            src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom1.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
            onClick={[Function]}
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Documentary
            </span>
            <span
              className="movie-card__year"
            >
              2015
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="/"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="/"
            onClick={[Function]}
          >
            Comedies
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="/"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
      </ul>
      <div>
        <div
          className="catalog__movies-list"
        />
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
